File: src/main/java/at/hrechny/predictionsbot/connector/apifootball/model/ApiFootballResponse.java
package at.hrechny.predictionsbot.connector.apifootball.model;

import java.util.List;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@ToString
public abstract class ApiFootballResponse<T> {

  private List<String> errors;
  private List<T> response;

}

File: src/main/kotlin/at/hrechny/predictionsbot/connector/impl/apifootball/model/ApiFootballResponse.kt
package at.hrechny.predictionsbot.connector.impl.apifootball.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
open class ApiFootballResponse<T> {
    var errors: List<String> = emptyList()
    var response: List<T>? = null
}
